src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1083,25 +1084,39 @@ private static void getElementsFromReferences(Set<TreeReference> referencesRemai
     /**
      * meant to be called after deserialization and initialization of handlers
      *
-     * @param newInstance true if the form is to be used for a new entry interaction,
-     *                    false if it is using an existing IDataModel
+     * @param formInitializationMode Specifies the mode in which the form is being initialized:
```

Comment:
Feels like this doc comment should be a link to the type so the values don't have to be inline here where they could become out of sync. I don't think the value comments are necessary since the names are descriptive. You could specifically call out that finalized_form_edit populates deprecatedid

Reply:
Done.